xbmc/utils/URIUtils.h

```diff
@@ -146,7 +146,7 @@ class URIUtils
   static bool IsMusicDb(const std::string& strFile);
   static bool IsNfs(const std::string& strFile);
   static bool IsOnDVD(const std::string& strFile);
-  static bool IsOnLAN(const std::string& strFile);
+  static bool IsOnLAN(const std::string& strFile, bool offLineCheck = false);
```

Comment:
I would suggest to not introduce new bool (default) param here, but a param of type of a new enum class `OfflineCheck`. Makes calling code much more readable, without having to look up the method declaration telling me what that `true`actually means.

Reply:
Yeah that's a good idea. Done in the last revision.
The diff is a bit bigger but it's well split by commits